test/test_kubeclient.rb

```diff
@@ -72,16 +72,28 @@ def test_exception
 
     client = Kubeclient::Client.new 'http://localhost:8080/api/'
 
-    exception = assert_raises(KubeException) do
+    exception = assert_raises(Kubeclient::HttpException) do
       service = client.create_service service
     end
 
-    assert_instance_of(KubeException, exception)
+    assert_instance_of(Kubeclient::HttpException, exception)
     assert_equal("converting  to : type names don't match (Pod, Namespace)",
                  exception.message)
     assert_equal(409, exception.error_code)
   end
 
+  def test_deprecated_exception
+    error_message = 'certificate verify failed'
+
+    stub_request(:get, 'http://localhost:8080/api')
+      .to_raise(OpenSSL::SSL::SSLError.new(error_message))
```

Comment:
I added this test to say "if an ssl error happens, kubeclient should react this way" ... I it wanted it to not matter what plumbing layer is in-between.